repos/system_upgrade/el7toel8/actors/restrictedpcisscanner/libraries/restrictedpcisscanner.py

```diff
@@ -0,0 +1,137 @@
+except ImportError:
+    # python2
+    JSONDecodeError = ValueError
+    FileNotFoundError = IOError  # pylint: disable=redefined-builtin
+
+REQUEST_TIMEOUT = 0.5  # TODO Consider moving it to the leapp env var
+UNSUPPORTED_DRIVER_NAMES_GLOBAL_FILE = os.path.join(
+    "/etc/leapp/files", "unsupported_driver_names.json"
+)
+UNSUPPORTED_PCI_IDS_GLOBAL_FILE = os.path.join(
+    "/etc/leapp/files", "unsupported_pci_ids.json"
+)
+
+# TODO Once the microservice for providing the data about removed hardware
+#   will be provisioned to the Red Hat costumers, then this variable default
+#   value should be overwritten
+PCIS_HOST_DEFAULT = ""
+API_VERSION = "v1"
+
+
+def get_restricted_pcis(http, update_fallback_data=False):
+    """
+    Get data about restricted driver names and unsupported pci_ids online.
+
+    The function makes API call to the web service, which provides
+    these data.
+
+    :param urllib3.PoolManager http: pool manager instance
+    :param bool update_fallback_data: if True writes the received data to
+        the corresponding json data files (inside ../data dir)
+
+    :returns: Tuple[dict, dict]
+    """
+    # TODO this constant should be above the scope of this func.
+    #   However due to the way how actor tested with CurrentActorMocked
+    #   we can't do this.
+    API_URL = "http://{host_port}/api/{api_version}/".format(
+        host_port=get_env(
+            "LEAPP_DEVEL_PCIS_HOST",
+            default=PCIS_HOST_DEFAULT,
+        ),
+        api_version=API_VERSION,
+    )
+    unsupported_driver_names = http.request(
+        "GET",
+        API_URL + "unsupported_driver_names/",
+        timeout=REQUEST_TIMEOUT,
+    )
+    unsupported_pci_ids = http.request(
+        "GET",
+        API_URL + "unsupported_pci_ids/",
+        timeout=REQUEST_TIMEOUT,
+    )
+
+    unsupported_driver_names = json.loads(
+        unsupported_driver_names.data.decode("utf-8")
+    )
+    unsupported_pci_ids = json.loads(unsupported_pci_ids.data.decode("utf-8"))
+    return unsupported_driver_names, unsupported_pci_ids
+
+
+def get_restricted_pcis_offline(driver_names_file, pci_ids_file, _open=open):
+    """
+    Get data about restricted driver names and unsupported pci_ids offline.
+
+    The function takes the data from locally stored json files in the given
+    location.
+
+    :param _open: made to make the testing easy
+    """
+    with _open(driver_names_file) as f:
+        unsupported_driver_names = json.load(f)
+    with _open(pci_ids_file) as f:
+        unsupported_pci_ids = json.load(f)
+    return unsupported_driver_names, unsupported_pci_ids
+
+
+def produce_restricted_pcis():
+    """
+    Produce RestrictedPCIDevice message from the online or offline sources.
+
+    The data sources preference order is the following:
+    1. We try to get the data from the /etc/leapp/files
+    2. We try to get the data from the only web service
+    3. Fallback mechanism, which provides the data from the data, stored inside
```

Comment:
If there is no fallback anymore - the comment doesn't make sense

Reply:
resolved in https://github.com/oamg/leapp-repository/pull/602/commits/06e630c346b953bbe1dc9d26628be33ae194abfa